src/EPR.Calculator.API/Controllers/DefaultParameterSettingController.cs

```diff
@@ -61,15 +63,45 @@ public IActionResult Create([FromBody] CreateDefaultParameterSettingDto createDe
                     return StatusCode(StatusCodes.Status500InternalServerError, exception);
                 }
             }
-                
+
             return new ObjectResult(null) { StatusCode = StatusCodes.Status201Created };
         }
 
         [HttpGet]
-        [Route("api/defaultParameterSetting")]
-        public IActionResult Get()
+        [Route("api/defaultParameterSetting/{parameterYear}")]
+        public IActionResult Get([FromRoute] string parameterYear)
         {
-            return Ok(new {Test = "This is not implemented!"});
+            if (string.IsNullOrEmpty(parameterYear))
+            {
+                // Return 400 error if the year is null or empty
+                return new ObjectResult("The input value cannot be null or empty. Please provide a valid value and try again") { StatusCode = StatusCodes.Status400BadRequest };
+            }
+
+            // Fetch the data based on the year passed to the API
+            var currentDefaultSetting = _context.DefaultParameterSettings
+                .SingleOrDefault(x => x.EffectiveTo == null && x.ParameterYear == parameterYear);
+
+            if (currentDefaultSetting == null)
+            {
+                // Return 404 error if the year is not found
+                return new ObjectResult("No data available for the specified year. Please check the year and try again.") { StatusCode = StatusCodes.Status404NotFound };
+            }
+
+            try
+            {
+                var _pramSettingDetails = _context.DefaultParameterSettingDetail.ToList();
+                var _templateDetails = _context.DefaultParameterTemplateMasterList;
+                var schemeParameters = CreateDefaultParameterSettingMapper.Map(currentDefaultSetting, _templateDetails);
+
+                // Return data in JSON format
+                var json = JsonConvert.SerializeObject(schemeParameters, Formatting.Indented);
```

Comment:
You do not need to do this. You can send the response as it is

Reply:
removed that